src/vivarium_gates_mncnh/components/observers.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any
+
+import pandas as pd
+from vivarium.framework.engine import Builder
+from vivarium.framework.results import Observer
+from vivarium_public_health.results import COLUMNS, PublicHealthObserver
+from vivarium_public_health.results import ResultsStratifier as ResultsStratifier_
+
+from vivarium_gates_mncnh.constants.data_values import PREGNANCY_OUTCOMES
+
+
+class ResultsStratifier(ResultsStratifier_):
+    def setup(self, builder: Builder) -> None:
+        self.age_bins = self.get_age_bins(builder)
+        # self.start_year = builder.configuration.time.start
+        # self.end_year = builder.configuration.time.end.year
+
+        self.register_stratifications(builder)
+
+    def register_stratifications(self, builder: Builder) -> None:
+        builder.results.register_stratification(
+            "age_group",
+            self.age_bins["age_group_name"].to_list(),
+            mapper=self.map_age_groups,
+            is_vectorized=True,
+            requires_columns=["age"],
+        )
+
+        builder.results.register_stratification(
+            "pregnancy_outcome",
+            list(
+                [
+                    PREGNANCY_OUTCOMES.STILLBIRTH_OUTCOME,
+                    PREGNANCY_OUTCOMES.LIVE_BIRTH_OUTCOME,
+                    PREGNANCY_OUTCOMES.PARTIAL_TERM_OUTCOME,
+                ]
+            ),
+            requires_columns=["pregnancy_outcome"],
+        )
+
+        builder.results.register_stratification(
+            "sex", ["Female", "Male"], requires_columns=["sex"]
+        )
+
+
+class BirthObserver(Observer):
```

Comment:
Do we actually need to record the individual birth outcomes to disk, since we will be using them in the same simulation during the next time-step?

Reply:
I don't think so but it was pretty easy to move it over and we don't have to use it in any runs. I can also remove it.